odl/solvers/functional/default_functionals.py

```diff
@@ -926,13 +1044,25 @@ def __init__(self, space):
             space, lower=0, upper=None)
 
     def __repr__(self):
-        """Return ``repr(self)``."""
-        return '{}({!r})'.format(self.__class__.__name__, self.domain)
+        """Return ``repr(self)``.
+
+        Examples
+        --------
+        >>> space = odl.rn(3)
+        >>> func = odl.solvers.IndicatorNonnegativity(space)
+        >>> func
+        IndicatorNonnegativity(rn(3))
+        """
+        posargs = [self.domain]
+        with npy_printoptions(precision=REPR_PRECISION):
+            inner_parts = signature_string_parts(posargs, [])
```

Comment:
Short thought here, shouldn't the "[]" be a default?

Reply:
I guess I could make that the default.